diff -uNr cmake-2.8.4/Modules/Platform/Darwin.cmake cmake-2.8.4.osx/Modules/Platform/Darwin.cmake
--- cmake-2.8.4/Modules/Platform/Darwin.cmake	2011-02-16 02:47:27.000000000 +0900
+++ cmake-2.8.4.osx/Modules/Platform/Darwin.cmake	2011-04-28 00:26:03.000000000 +0900
@@ -42,8 +42,8 @@
 ENDIF(HAVE_FLAG_SEARCH_PATHS_FIRST)
 
 SET(CMAKE_PLATFORM_HAS_INSTALLNAME 1)
-SET(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-dynamiclib -Wl,-headerpad_max_install_names")
-SET(CMAKE_SHARED_MODULE_CREATE_C_FLAGS "-bundle -Wl,-headerpad_max_install_names")
+SET(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-dynamiclib -single_module -flat_namespace -undefined dynamic_lookup")
+SET(CMAKE_SHARED_MODULE_CREATE_C_FLAGS "-bundle -flat_namespace -undefined dynamic_lookup")
 SET(CMAKE_SHARED_MODULE_LOADER_C_FLAG "-Wl,-bundle_loader,")
 SET(CMAKE_SHARED_MODULE_LOADER_CXX_FLAG "-Wl,-bundle_loader,")
 SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dylib" ".so" ".a")
