diff -uNr Cython-0.15.1.orig/Demos/embed/Makefile Cython-0.15.1/Demos/embed/Makefile
--- Cython-0.15.1.orig/Demos/embed/Makefile	2011-10-21 20:30:48.000000000 +0900
+++ Cython-0.15.1/Demos/embed/Makefile	2011-10-21 20:31:46.000000000 +0900
@@ -14,7 +14,7 @@
 SYSLIBS= $(shell $(PYTHON) -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('SYSLIBS'))")
 
 embedded: embedded.o
-	$(LINKCC) -o $@ $^ -L$(LIBDIR1) -L$(LIBDIR2) -l$(PYLIB) $(LIBS) $(SYSLIBS) $(LINKFORSHARED)
+	$(LINKCC) -o $@ $^ -lPython -L$(LIBDIR1) -L$(LIBDIR2) -l$(PYLIB) $(LIBS) $(SYSLIBS) $(LINKFORSHARED)
 
 embedded.o: embedded.c
 	$(CC) -c $^ -I$(INCDIR)
