diff -uNr ganglia-3.1.7.orig/configure ganglia-3.1.7/configure
--- ganglia-3.1.7.orig/configure	2010-02-17 20:05:39.000000000 +0900
+++ ganglia-3.1.7/configure	2012-03-16 03:46:50.000000000 +0900
@@ -25260,7 +25260,7 @@
 fi
 
 
-varstatedir="/var/lib"
+varstatedir="${prefix}/var/lib"
 
 # Used when static linking requested
 EXPORT_SYMBOLS="-export-dynamic"
@@ -25453,7 +25453,7 @@
 #define SUPPORT_GEXEC 0
 _ACEOF
 
-		varstatedir="/var/db"
+		varstatedir="${prefix}/var/db"
 		osreldate=`awk '/^\#define[[:space:]]*__FreeBSD_version/ { print $3 }' /usr/include/osreldate.h`
 		if test $osreldate -lt 502102; then
 		   CFLAGS="-pthread $CFLAGS -D_REENTRANT"
@@ -25470,7 +25470,7 @@
 #define SUPPORT_GEXEC 0
 _ACEOF
 
-		varstatedir="/var/db"
+		varstatedir="${prefix}/var/db"
 
 cat >>confdefs.h <<\_ACEOF
 #define DRAGONFLYBSD 1
@@ -25482,7 +25482,7 @@
 #define SUPPORT_GEXEC 0
 _ACEOF
 
-		varstatedir="/var/db"
+		varstatedir="${prefix}/var/db"
 
 cat >>confdefs.h <<\_ACEOF
 #define NETBSD 1
@@ -25494,7 +25494,7 @@
 #define SUPPORT_GEXEC 0
 _ACEOF
 
-		varstatedir="/var/db"
+		varstatedir="${prefix}/var/db"
 
 cat >>confdefs.h <<\_ACEOF
 #define OPENBSD 1
diff -uNr ganglia-3.1.7.orig/contrib/ganglia-rrd-modify.pl ganglia-3.1.7/contrib/ganglia-rrd-modify.pl
--- ganglia-3.1.7.orig/contrib/ganglia-rrd-modify.pl	2010-02-17 20:05:39.000000000 +0900
+++ ganglia-3.1.7/contrib/ganglia-rrd-modify.pl	2012-03-16 03:49:42.000000000 +0900
@@ -20,8 +20,8 @@
 my $process_name = basename $0;
 
 # Define some useful variables:
-my $rrd_dir = '/var/lib/ganglia/rrds';
-my $rrdtool_path = '/usr/bin/rrdtool';
+my $rrd_dir = '/usr/osxws/var/lib/ganglia/rrds';
+my $rrdtool_path = '/usr/osxws/bin/rrdtool';
 my $heartbeat = 0;
 my $grow = 0;
 my $shrink = 0;
diff -uNr ganglia-3.1.7.orig/gmetad/conf.c ganglia-3.1.7/gmetad/conf.c
--- ganglia-3.1.7.orig/gmetad/conf.c	2010-02-17 20:05:39.000000000 +0900
+++ ganglia-3.1.7/gmetad/conf.c	2012-03-16 03:48:52.000000000 +0900
@@ -275,7 +275,7 @@
    config->debug_level = 0;
    config->should_setuid = 1;
    config->setuid_username = "nobody";
-   config->rrd_rootdir = "/var/lib/ganglia/rrds";
+   config->rrd_rootdir = "/usr/osxws/var/lib/ganglia/rrds";
    config->scalable_mode = 1;
    config->all_trusted = 0;
    config->num_RRAs = 5;
diff -uNr ganglia-3.1.7.orig/gmond/modules/python/README ganglia-3.1.7/gmond/modules/python/README
--- ganglia-3.1.7.orig/gmond/modules/python/README	2010-02-17 20:05:39.000000000 +0900
+++ ganglia-3.1.7/gmond/modules/python/README	2012-03-16 03:50:51.000000000 +0900
@@ -101,11 +101,11 @@
     module {
       name = "python_module"
       path = "modpython.so"
-      params = "/usr/lib/ganglia/python_modules"
+      params = "/usr/osxws/lib/ganglia/python_modules"
     }
   }
 
-  include ('/etc/ganglia/conf.d/*.pyconf')
+  include ('/usr/osxws/etc/ganglia/conf.d/*.pyconf')
 
 The most significant part of this configuration is the 'params'
 directive. The path that has been assigned to this directive will
diff -uNr ganglia-3.1.7.orig/web/Makefile ganglia-3.1.7/web/Makefile
--- ganglia-3.1.7.orig/web/Makefile	2010-02-17 20:05:39.000000000 +0900
+++ ganglia-3.1.7/web/Makefile	2012-03-16 03:48:04.000000000 +0900
@@ -176,7 +176,7 @@
 target_alias = 
 top_builddir = ..
 top_srcdir = ..
-varstatedir = /var/lib
+varstatedir = ${prefix}/var/lib
 FIXCONFIG = $(top_srcdir)/scripts/fixconfig
 all: all-am
 
