diff -uNr git-1.7.3.5.orig/perl/Makefile git-1.7.3.5/perl/Makefile
--- git-1.7.3.5.orig/perl/Makefile	2011-01-08 15:58:55.000000000 +0900
+++ git-1.7.3.5/perl/Makefile	2011-01-08 16:23:32.000000000 +0900
@@ -38,7 +38,7 @@
 	echo '	echo $(instdir_SQ)' >> $@
 else
 $(makfile): Makefile.PL ../GIT-CFLAGS
-	$(PERL_PATH) $< PREFIX='$(prefix_SQ)' INSTALL_BASE=''
+	$(PERL_PATH) $< SITEPREFIX='$(prefix_SQ)' INSTALL_BASE=''
 endif
 
 # this is just added comfort for calling make directly in perl dir
